### PR TITLE
Solaris/Illumos build implementation proposal.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: libhfcommon/log.c [hunk 1]
```diff
@@ -181,3 +181,18 @@ int logFd(void) {
...
+    va_list ap;
+    va_start(ap, fmt);
+    vdprintf(fd, fmt, ap);
+    va_end(ap);
+}
+
+int vdprintf(int fd, const char *fmt, va_list ap) {
+    FILE *fp;
+    fp = fdopen(fd, "w");
+    return vfprintf(fp, fmt, ap);
```
Comment (robertswiecki) [inline libhfcommon/log.c:196]: Is fclose needed here?

**Review by robertswiecki — CHANGES_REQUESTED** [review]

File: libhfcommon/log.c (hunk 1 above)
Comment (robertswiecki) [reply libhfcommon/log.c:196]: Sorry, I mean fclose would close the fd as well, but aren't we allocating struct for FILE* each time we do fdopen?

Comment (robertswiecki) [reply libhfcommon/log.c:196]: May fdopen/fclose on dup'd fd?

Comment (devnexen) [reply libhfcommon/log.c:196]: tried to avoid using the opaque filling approach here but ok may need fclose indeed

Comment (devnexen) [reply libhfcommon/log.c:196]: tried but that faults.

Comment (devnexen) [reply libhfcommon/log.c:196]: will try using the opaque field instead.

**Comment by robertswiecki** [comment]
Thanks!

**Pull request merged**